mods/ra/maps/soviet-06a/soviet06a.lua

```diff
@@ -0,0 +1,150 @@
+CameraBarrierTrigger = { CPos.New(65, 39), CPos.New(65, 40), CPos.New(66, 40), CPos.New(66, 41), CPos.New(67, 41), CPos.New(67, 42), CPos.New(68, 42), CPos.New(68, 43), CPos.New(68, 44) }
+CameraBaseTrigger = { CPos.New(53, 42), CPos.New(54, 42), CPos.New(54, 41), CPos.New(55, 41), CPos.New(56, 41), CPos.New(56, 40), CPos.New(57, 40), CPos.New(57, 39), CPos.New(58, 39), CPos.New(59, 39), CPos.New(59, 38), CPos.New(60, 38), CPos.New(61, 38) }
+
+Trigger.OnEnteredFootprint(TruckGoalTrigger, function(a, id)
+	if not truckGoalTrigger and a.Owner == player and a.Type == "truk" then
+		truckGoalTrigger = true
+		player.MarkCompletedObjective(sovietObjective)
+	end
+end)
+
+Trigger.OnEnteredFootprint(CameraBarrierTrigger, function(a, id)
+	if not cameraBarrierTrigger and a.Owner == player then
+		cameraBarrierTrigger = true
+		local cameraBarrier = Actor.Create("camera", true, { Owner = player, Location = CameraBarrier.Location })
+		Trigger.AfterDelay(DateTime.Seconds(15), function()
+			cameraBarrier.Kill()
+		end)
+	end
+end)
+
+Trigger.OnEnteredFootprint(CameraBaseTrigger, function(a, id)
+	if not cameraBaseTrigger and a.Owner == player then
+		cameraBaseTrigger = true
+		local cameraBase1 = Actor.Create("camera", true, { Owner = player, Location = CameraBase1.Location })
+		local cameraBase2 = Actor.Create("camera", true, { Owner = player, Location = CameraBase2.Location })
+		local cameraBase3 = Actor.Create("camera", true, { Owner = player, Location = CameraBase3.Location })
+		local cameraBase4 = Actor.Create("camera", true, { Owner = player, Location = CameraBase4.Location })
+		Trigger.AfterDelay(DateTime.Minutes(1), function()
+			cameraBase1.Kill()
+			cameraBase2.Kill()
+			cameraBase3.Kill()
+			cameraBase4.Kill()
+		end)
+	end
+end)
+
+Trigger.OnAllKilled(Trucks, function()
+	if not controlCenterTrigger then
+		enemy.MarkCompletedObjective(alliedObjective)
+	end
+end)
+
+Trigger.OnKilled(Apwr, function(building)
+	BaseBuildings[1][4] = false
+end)
+
+Trigger.OnKilled(Barr, function(building)
+	BaseBuildings[2][4] = false
+end)
+
+Trigger.OnKilled(Proc, function(building)
+	BaseBuildings[3][4] = false
+end)
+
+Trigger.OnKilled(Weap, function(building)
+	BaseBuildings[4][4] = false
+end)
+
+Trigger.OnKilled(Apwr2, function(building)
+	BaseBuildings[5][4] = false
+end)
+
+-- Activate the AI once the player deployed the Mcv
+Trigger.OnRemovedFromWorld(Mcv, function()
+	if not mcvDeployed then
+		mcvDeployed = true
+		BuildBase()
+		Trigger.AfterDelay(DateTime.Minutes(1), ProduceInfantry)
+		Trigger.AfterDelay(DateTime.Minutes(2), ProduceArmor)
+		Trigger.AfterDelay(DateTime.Minutes(2), function()
+			Utils.Do(BaseAttackers, function(actor)
+				IdleHunt(actor)
+			end)
+		end)
+	end
+end)
+
+IdleHunt = function(actor)
```

Comment:
This shouldn't be necessary as you can access the `IdleHunt` function from `soviet06a-AI.lua`.